source/extensions/filters/common/expr/context.cc

```diff
@@ -139,6 +139,20 @@ absl::optional<CelValue> ResponseWrapper::operator[](CelValue key) const {
     return CelValue::CreateMap(&trailers_);
   } else if (value == Flags) {
     return CelValue::CreateInt64(info_.responseFlags());
+  } else if (value == GrpcStatus) {
+    if (trailers_.value_ != nullptr && trailers_.value_->GrpcStatus() != nullptr) {
+      int64_t status;
+      if (absl::SimpleAtoi(trailers_.value_->GrpcStatus()->value().getStringView(), &status)) {
+        return CelValue::CreateInt64(status);
+      }
+    }
+    if (headers_.value_ != nullptr && headers_.value_->GrpcStatus() != nullptr) {
+      int64_t status;
+      if (absl::SimpleAtoi(headers_.value_->GrpcStatus()->value().getStringView(), &status)) {
+        return CelValue::CreateInt64(status);
+      }
+    }
+    return {};
```

Comment:
Do you have a test case that covers no headers/trailers and/or bad header/trailer values? (Optimally you could use `Common::getGrpcStatus` but not sure if it's worth it or not.

Reply:
Thanks for the pointer to the `Common:getGrpcStatus` bit. I did not know that existed. That led me to look at the access log code that uses it, which ultimately led me to expand the fallbacks.

Now, after looking in trailers and headers, this will attempt to look at http response code and finally return `Unknown` if not found.

Matching the logging behavior seemed important.

I added a bunch of test cases for this updated behavior to match.  Please let me know what you think.

Reply:
Thanks, is there is any reasonable way to unify this logic and the access log logic so they don't fall out of sync? I agree they should be the same.

/wait-any